hypervisor/dm/vpci/msix.c

```diff
@@ -69,7 +69,7 @@ static int vmsix_remap_entry(struct pci_vdev *vdev, uint32_t index, bool enable)
 	 * fields with a single QWORD write, but some hardware can accept 32 bits
 	 * write only
 	 */
-	mmio_write32((uint32_t)(info.pmsi_addr), (const void *)&(pentry->addr));
+	mmio_write32((uint32_t)info.pmsi_addr, (const void *)&(pentry->addr));
 	mmio_write32((uint32_t)(info.pmsi_addr >> 32U), (const void *)((char *)&(pentry->addr) + 4U));
```

Comment:
I see a few patches are adding () to x->y, we need to clarify which one is the right solution. OItherwise, it is just back and forth...

Reply:
In this case, with '()', MISRA-C scan reports violation: "Value is not of appropriate type. :  (unsigned int and unsigned long)", and the violation was gone without it.

I can do some study for this.